src/pytest_postgresql/factories.py

```diff
@@ -133,12 +133,15 @@ def drop_postgresql_database(user, host, port, db, version):
     conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
     cur = conn.cursor()
     # We cannot drop the database while there are connections to it, so we
-    # terminate all connections first.
+    # terminate all connections first while not allowing new connections.
     if float(version) >= 9.2:
         pid_column = 'pid'
     else:
         pid_column = 'procpid'
     cur.execute(
+        'UPDATE pg_database SET datallowconn=false WHERE datname = %s;',
```

Comment:
I don't see why a per-database setting would persist after the database is dropped. It can be checked by using the database fixture in two tests.